static/js/timerender.ts

```diff
@@ -24,6 +24,20 @@ export function clear_for_testing(): void {
     next_timerender_id = 0;
 }
 
+// by providing a default string of 'PP' or any of its variants for `formatStr`
+// it will format dates in whichever way is appropriate to the locale
+function localizedFormat(date: Date | number, formatOptions?: Intl.DateTimeFormatOptions): string {
+    const defaultFormatOptions: Intl.DateTimeFormatOptions = {
+        month: "short",
+        day: "2-digit",
+        year: "numeric",
+    };
+    const currentLocale = user_settings.default_language;
```

Comment:
What happens if the `Intl` library's idea of what locales exist is different from Zulip's? 

(Not sure this is super plausible, but I suppose the simplest situation would be if `user_settings.default_language` was just an invalid locale string; we could just write a test for this with `user_settings.default_language="invalid"` if there isn't one already)

Reply:
Nice catch, thanks. We should definitely add some tests for this case and possibly include some fallbacks if it's not working correctly now.

Reply:
According to [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat#using_datetimeformat) in `Intl` it should use default locale. But i think it would be safer to have some test cases, so I will write them.

Reply:
After some tests written and failed I added `try {} catch {}`, to provide 'default' value to `Intl.DateTimeFormat` function.

Reply:
Sounds good!